libevmasm/RuleList.h

```diff
@@ -162,6 +160,25 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleList(
 		{{Instruction::OR, {X, {Instruction::NOT, {X}}}}, [=]{ return ~u256(0); }, true},
 		{{Instruction::OR, {{Instruction::NOT, {X}}, X}}, [=]{ return ~u256(0); }, true},
 	};
+}
+
+
+/// @returns a list of simplification rules given certain match placeholders.
+/// A, B and C should represent constants, X and Y arbitrary expressions.
+/// The simplifications should never change the order of evaluation of
+/// arbitrary operations.
+template <class Pattern>
+std::vector<SimplificationRule<Pattern>> simplificationRuleList(
+	Pattern A,
+	Pattern B,
+	Pattern C,
+	Pattern X,
+	Pattern Y
+)
+{
+	std::vector<SimplificationRule<Pattern>> rules;
+
+	rules += simplificationRuleListPart1(A, B, C, X, Y);
 
 	// Replace MOD X, <power-of-two> with AND X, <power-of-two> - 1
```

Comment:
Just for clearity, perhaps split out the rest of this function into `..Part2` and then call only `...Part1` and `...Part2` here?